### PR TITLE
Add torch.argsort mirroring similar functionality in numpy.

### PR DESCRIPTION
**Pull request by resistor**
Per issue #9542

### CODE REVIEW

File: torch/functional.py [hunk 1]
```diff
@@ -403,3 +404,36 @@ def argmin(input, dim=None, keepdim=False):
...
+
+
+        >>> torch.argsort(a, dim=1)
+        tensor([[2, 0, 3, 1],
+                [3, 2, 1, 0],
+                [2, 1, 0, 3],
+                [3, 2, 1, 0]])
+    """
+    if dim is None:
+        return torch.sort(input, input.dim() - 1, descending)[1]
```
Comment (vadimkantorov) [inline torch/functional.py]: `input.dim() - 1` can be just `-1` (more idiomatic)?

File: torch/functional.py [hunk 2]
```diff
@@ -403,3 +404,36 @@ def argmin(input, dim=None, keepdim=False):
     if dim is None:
         return torch._argmin(input.contiguous().view(-1), dim=0, keepdim=False)
     return torch._argmin(input, dim, keepdim)
+
+
+def argsort(input, dim=None, descending=False):
```
Comment (vishwakftw) [inline torch/functional.py]: I think this should be just as simple as:
```
def argsort(input, *args):
    return torch.sort(input, *args)[1]
```

Comment (resistor) [reply torch/functional.py]: argsort does not accept the output parameter that torch.sort does, so this doesn't work.

Comment (vishwakftw) [reply torch/functional.py]: Oh yes, you're right.

**Review by goldsborough — APPROVED** [review]

**Pull request closed without merging**